### ISSUE TITLE
Bug: [strict-boolean-expressions] should flag `true | undefined` as a nullable boolean.

### ISSUE THREAD

**Issue by kirkwaiblinger** [typescript-eslint/typescript-eslint#10859]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.2.2&fileType=.tsx&code=CYUwxgNghgTiAEYD2A7AzgF3hgXNmArggD7wEqgBmAliiMANwBQTA9K-GgBZIETDxK0AOZNqleAAoMASngBvFgF8WoSLATJ0WSnkpQIaEmQogadRi3bxgSEGkEixEyZTmKmSoA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6ZfKsugIwHs%2BSAIZN6AD2LQUySnyaoMkAO5DoTSODABfEFqA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
declare const t: true | undefined;

if (t) {}
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/strict-boolean-expressions": "warn"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

I expect that the conditional should be flagged as a nullable boolean.

### Actual Result

no report.

### Additional Info

By contrast, `false | undefined` and `boolean | undefined` do trigger the rule.

**Comment by bradzacher** [comment]
The reason the rule doesn't flag this is because the type has just one truthy value and one falsey value. So it is clear what `if (foo)` means - there's no ambiguity.

This is also why the false version does flag - because it now has two falsey values. This means that `if (!foo)` is ambiguous.

So yeah this is fully intentional.

**Comment by kirkwaiblinger** [comment]
I suppose that's fair.... But annoyingly, it means there is no way to configure the rule to ban this even if you want to (unlike, for example, `allowNullableObject: false`, which is also equally "clear" in a sense but I still want to ban 😀)

**Comment by kirkwaiblinger** [comment]
In other words, I'm still thwarted in trying to mandate I only get _exactly_ the values `true` or `false` at runtime.

**Comment by JoshuaKGoldberg** [comment]
Would it be worth it to add a new value to the existing [`allowNullableBoolean`](https://typescript-eslint.io/rules/strict-boolean-expressions/#allownullableboolean) option, maybe `'boolean-only'`?

**Comment by bradzacher** [comment]
I'd personally say no -- but worth leaving as evaluating engagment?

**Comment by ronami** [comment]
I think I'm -0.5 on this.

I like the idea of getting exactly booleans in conditionals. However, as @brad mentioned, there's no ambiguity, and unless reporting this can fix a potential bug/issue, the report mainly creates noise.

👍 for leaving this as evaluating community engagement.

**Comment by JoshuaKGoldberg** [comment]
It's been about 6 months since this was marked as `evaluating community engagement`. In that time, nobody's expressed any text opinions in favor of this change. We'll go ahead and close it out as non-actionable for typescript-eslint itself.

If anybody wants to fork the existing rule and publish it as a standalone plugin, please do! 
* https://typescript-eslint.io/developers/custom-rules: shows how to write custom rules
* https://typescript-eslint.io/developers/eslint-plugins: shows how to package them as plugins

Cheers! 💙
